libdnf/dnf-transaction.cpp

```diff
@@ -1431,17 +1432,20 @@ dnf_transaction_commit(DnfTransaction *transaction, HyGoal goal, DnfState *state
     if (!dnf_context_plugin_hook(priv->context, PLUGIN_HOOK_ID_CONTEXT_PRE_TRANSACTION, &data, nullptr))
         goto out;
 
-    // FIXME get commandline
-    if (sack) {
-        rpmdb_begin = dnf_sack_get_rpmdb_version(sack);
-    } else {
-        // if sack is not available, create a custom instance
-        rpmdb_version_sack = dnf_sack_new();
-        dnf_sack_load_system_repo(rpmdb_version_sack, nullptr, DNF_SACK_LOAD_FLAG_NONE, nullptr);
-        rpmdb_begin = dnf_sack_get_rpmdb_version(rpmdb_version_sack);
-        g_object_unref(rpmdb_version_sack);
+    // Open rpm database if it is not already open
+    if (!rpmtsGetRdb(priv->ts)) {
+        rc = rpmtsOpenDB(priv->ts, rpmtsGetDBMode(priv->ts));
+        if (rc != 0) {
+            ret = FALSE;
+            g_set_error(
+                error, DNF_ERROR, DNF_ERROR_INTERNAL_ERROR, _("Error %i opening rpm database"), rc);
+            goto out;
+        }
     }
-    swdb->beginTransaction(_get_current_time(), rpmdb_begin, "", priv->uid);
+
+    rpmdb_begin.reset(rpmdbCookie(rpmtsGetRdb(priv->ts)));
+    // FIXME get commandline
```

Comment:
Please can you also report when `rpmdbCookie(rpmtsGetRdb(priv->ts))` returns `NULL`?

Reply:
OK.